fix : Fixed compilation errors

CoffeeSwt opened the pull request:
The constructor can't use the override keyword, so I rewrote it, and the config.cpp file must contain CfgPatches or it will compile with errors, which I added to the end.

CoffeeSwt committed:
fix : Fixed compilation errors

kuzyn1 commented:
> The constructor can't use the override keyword, so I rewrote it, and the config.cpp file must contain CfgPatches or it will compile with errors, which I added to the end.

Hey, could you explain how to structure the files exactly in this mod? Even with your fix I cant get rid of compilation errors.

In the main folder I have my config.cpp with CfgPatches and CfgMods with my custom class for this mod.
In Data\DynamicSounds I have another config.cpp with my single shader and soundset. Also ogg files are included here. Then in Scripts\4_World\SoundHandler I have AmbientSound.c, exactly like you with your fixes but 1 my own territory inserted.

And from that I make a new pbo and put it in the folder with my other mods like map.
When I try to run this on a server I get an error:

![image](https://github.com/user-attachments/assets/3c7f5555-ffbb-4d44-844a-32175daf1f49)

CoffeeSwt commented:
> > The constructor can't use the override keyword, so I rewrote it, and the config.cpp file must contain CfgPatches or it will compile with errors, which I added to the end.
> 
> Hey, could you explain how to structure the files exactly in this mod? Even with your fix I cant get rid of compilation errors.
> 
> In the main folder I have my config.cpp with CfgPatches and CfgMods with my custom class for this mod. In Data\DynamicSounds I have another config.cpp with my single shader and soundset. Also ogg files are included here. Then in Scripts\4_World\SoundHandler I have AmbientSound.c, exactly like you with your fixes but 1 my own territory inserted.
> 
> And from that I make a new pbo and put it in the folder with my other mods like map. When I try to run this on a server I get an error:
> 
> ![image](https://private-user-images.githubusercontent.com/4819951/400262537-3c7f5555-ffbb-4d44-844a-32175daf1f49.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MzQ4OTMsIm5iZiI6MTczNjUzNDU5MywicGF0aCI6Ii80ODE5OTUxLzQwMDI2MjUzNy0zYzdmNTU1NS1mZmJiLTRkNDQtODQ0YS0zMjE3NWRhZjFmNDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTBUMTg0MzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQ5YWQ0NWY3N2NlZmZkYWY1NjU4ZmYxNzk0MTE1Zjc4ZGZhOThlNTQyNmY2YzExMmI3YWMwODE0MGU3OWQ4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.efMrnWcs3JAkDXCus_B39nbrUO33P4WFCV51153WN-U)

It seems that you dont have the basic mod of Dynamic Ambient Sound System, so you can't find the class AmbientHandler.
Make sure you have load the Dynamic Ambient Sound System before the Custom_DSS.
```bash
-mod=@Dynamic Ambient Sound System;@Custom_DSS;
```
Dynamic Ambient Sound System https://steamcommunity.com/sharedfiles/filedetails/?id=3302539491

tips: Use the tools box of mikero to figure out the compilation errors.
https://mikero.bytex.digital/Downloads
And here is a simple guide of how to use it:
https://www.youtube.com/watch?v=TglUj7ekkdo

kuzyn1 commented:
@CoffeeSwt thank you for explaining, for some reason I thought that this mod is a "standalone" version.

AmKoma commented:
Thanks for fixxing. @CoffeeSwt
